packages/client/katello-pull-provider-migrate/katello-pull-provider-migrate

```diff
@@ -0,0 +1,80 @@
+#!/bin/sh
+
+# set SBINDIR to /sbin if it is not set
+if [ -z "$SBINDIR" ]; then
+    SBINDIR=/usr/sbin
+fi
+
+# check if system is registered
+subscription-manager identity > /dev/null
+if [ $? -ne 0 ]; then
+    echo "Please register with subscription-manager first, then re-run `$SBINDIR/katello-pull-provider-migrate`."
+    exit 1
+fi
+
+echo "Getting configuration from subscription-manager..."
+KPPTEMPFILE=$(mktemp kpp_tempfile_XXXXXXXX)
+subscription-manager config --list > $KPPTEMPFILE
+CONSUMER_CERT_DIR=$(grep 'consumercertdir' $KPPTEMPFILE | cut -d= -f2 | xargs | sed 's/[][]//g')
+CERT_FILE=$CONSUMER_CERT_DIR/cert.pem
+KEY_FILE=$CONSUMER_CERT_DIR/key.pem
+CA_FILE=$(grep 'repo_ca_cert = ' $KPPTEMPFILE | cut -d= -f2 | xargs | cut -d ' ' -f1)
+SERVER_NAME=$(grep 'hostname = ' $KPPTEMPFILE | cut -d= -f2 | xargs | cut -d ' ' -f1)
+
+# fail if no server name, cert dir, or ca file
+if [ -z "$SERVER_NAME" ] || [ -z "$CONSUMER_CERT_DIR" ] || [ -z "$CA_FILE" ]; then
+    echo "Unable to determine config from `subscription-manager config --list`; exiting"
+    exit 1
+fi
+
+# fail if client is not registered to a Katello
+if ! grep -q 'prefix = \/rhsm' $KPPTEMPFILE; then
+    echo "Client is not registered to Katello; exiting"
+    exit 2
+fi
+
+rm -f $KPPTEMPFILE
+
+# set SYSCONFDIR to /etc if it is not set
+if [ -z "$SYSCONFDIR" ]; then
+    SYSCONFDIR=/etc
+fi
+
+# see if /etc/yggdrasil/config.toml exists
```

Comment:
This will be a fun one, and is viable as a phase 2 addition (but should be tracked if it is). On RHEL, this won't be `yggdrasil` but rather `rhc` and you'll need to either include paths for both cases or add some conditional logic to determine which is present (rhc or yggdrasil) and set the paths based on this information.

Reply:
> viable as a phase 2 addition

Does this mean I don't have to worry about it just yet?

Reply:
Yea, I think you could focus on CentOS support to start with and move to RHEL support later.